### PR TITLE
feat: add support for negative numbers and numeric ranges

### PR DESCRIPTION
**Pull request by McManning**
Fixes an issue in `transformNumericFilters` where negative values were rejected - e.g. `discount<-10` and adds support for the numeric range format detailed here: https://www.algolia.com/doc/api-reference/api-parameters/numericFilters/#numeric-range

### CODE REVIEW

**Comment by McManning** [comment]
I had this silently ignore malformed numeric filter strings, which looked to be how the original behaviour worked. Is that still fine or would you like this to throw an exception?

**Comment by joemcelroy** [comment]
> I had this silently ignore malformed numeric filter strings, which looked to be how the original behaviour worked. Is that still fine or would you like this to throw an exception?

we should throw an exception here!

File: packages/searchkit/src/filters.ts [hunk 1]
```diff
@@ -20,19 +20,36 @@ export const transformNumericFilters = (
...
-      /([\w\.\_\-]+)(\=|\!\=|\>|\>\=|\<|\<\=)(\d+)/
+  return numericFilters.reduce((sum, filter: string) => {
+    let match, field, operator, value, maxValue = ''
+    let groups = filter.match(
+      /([\w\.\_\-]+)\s*(\=|\!\=|\>|\>\=|\<|\<\=)\s*(-?\d+)/
     )
 
+    if (groups) {
+      [match, field, operator, value] = groups
+      value
```
Comment (joemcelroy) [inline packages/searchkit/src/filters.ts:31]: `value` - is this intentional?

Comment (McManning) [reply packages/searchkit/src/filters.ts]: Sure wasn't, good catch

**Review by joemcelroy — APPROVED** [review]
Lgtm!

**Comment by joemcelroy** [comment]
released in 4.6.2

**Pull request merged**